testnet/stacks-node/src/run_loop/neon.rs

```diff
@@ -149,45 +155,66 @@ impl RunLoop {
 
         let mut coordinator_dispatcher = event_dispatcher.clone();
 
-        let (network, _) = self.config.burnchain.get_bitcoin_network();
-
-        let burnchain_config = match burnchain_opt {
-            Some(burnchain_config) => burnchain_config.clone(),
-            None => match Burnchain::new(
-                &self.config.get_burn_db_path(),
-                &self.config.burnchain.chain,
-                &network,
-            ) {
-                Ok(burnchain) => burnchain,
-                Err(e) => {
-                    error!("Failed to instantiate burnchain: {}", e);
-                    panic!()
-                }
-            },
-        };
         let chainstate_path = self.config.get_chainstate_path();
         let coordinator_burnchain_config = burnchain_config.clone();
 
-        thread::spawn(move || {
-            let mut boot_data =
-                ChainStateBootData::new(&coordinator_burnchain_config, initial_balances, None);
+        let first_block_height = burnchain_config.first_block_height as u128;
+        let pox_prepare_length = burnchain_config.pox_constants.prepare_length as u128;
+        let pox_reward_cycle_length = burnchain_config.pox_constants.reward_cycle_length as u128;
+        let pox_rejection_fraction = burnchain_config.pox_constants.pox_rejection_fraction as u128;
+
+        let boot_block = Box::new(move |clarity_tx: &mut ClarityTx| {
```

Comment:
It doesn't seem correct to me for this code to be outside the `chainstate` module -- I think it'd be better to do something like introduce an `initialize` method to coordinator, which we can run synchronously, to ensure that the coordinator is the first method to call `open`.

This should probably also change the watchdog to `connect`, rather than `open`

Reply:
How would you address https://github.com/blockstack/stacks-blockchain/issues/2066 with this approach?

Reply:
I think one of two things:

1. Not address it all. I'm not too worried about this being difficult to understand-- I think it's ultimately pretty easy to find where the boot block code is getting initialized, since there are relatively few calls to open_and_exec. We could also always add a debug! log to the ChainStateBootData object to log every time a closure is added if it makes debugging easier.
2. Move the code from the chain state coordinator into the StacksChainState code that inserts the boot contracts, so that it is executed always, this eliminates one of the closures.

Reply:
I'd be in favor of (2).  Part of the rationale for #2066 is that external parties may need to audit the genesis block -- i.e. for legal and tax compliance purposes (this has happened before in Stacks 1.0).  Making it straightforward to show auditors that the genesis block was correctly instantiated will significantly reduce our collective anxiety when assisting with these audits.

Reply:
Either way, I'm also worried about the race on `open()` -- open() should only be called once, and everywhere else that is currently calling `open` should switch to `connect` and panic if the database hasn't already been created.

Reply:
A similar panic behavior was already implemented. It's not at upfront at the method you're describing, but essentially, if the first call to open in not providing a `ChainStateBootData`, then you'll get a panic. That's how I was able to experience this bug on Bitcoin Testnet.

Reply:
Stashing some code for now and will address in https://github.com/blockstack/stacks-blockchain/issues/2095 so we can make some progress on Xenon deployments